quic/s2n-quic-transport/src/stream/controller.rs

```diff
@@ -352,10 +366,24 @@ impl OutgoingController {
         self.wake_unblocked();
     }
 
-    fn poll_open_stream(&mut self, context: &Context) -> Poll<()> {
+    fn poll_open_stream(
+        &mut self,
+        open_token: &mut connection::OpenToken,
+        context: &Context,
+    ) -> Poll<()> {
         if self.available_stream_capacity() < VarInt::from_u32(1) {
-            // Store a waker that can be woken when we get more credit
-            self.wakers.push(context.waker().clone());
+            if let Some(index) = open_token.index(&self.expired_token) {
+                let prev = &self.wakers[index];
+                // update the waker if it's changed
+                if !prev.will_wake(context.waker()) {
+                    self.wakers[index] = context.waker().clone();
```

Comment:
this is the case where the caller has called `poll_open_stream` repeatedly with the same connection handle? so we just overwrite the waker in that case since the connection handle only needs to be woken once

Reply:
that's exactly right